src/features/activity-log/components/activity-log/activity-log-timeline.tsx

```diff
@@ -0,0 +1,78 @@
+import { Card } from 'components/ui/card';
+import ActivityLogGroup from './activity-log-group';
+import { ActivityGroup } from './activity-log.types';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { debounce } from 'lodash';
+
+const ActivityLogTimeline = ({ activities }: { activities: ActivityGroup[] }) => {
+  const [visibleCount, setVisibleCount] = useState(5); 
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  // Memoize the handleScroll function with useCallback to include it in dependencies
+  const handleScroll = useCallback(
+    debounce(() => {
+      const container = containerRef.current;
+      if (container) {
+        if (container.scrollHeight - container.scrollTop <= container.clientHeight + 200) {
+          // Load more activities if not at the end
+          if (visibleCount < activities.length) {
+            setVisibleCount((prev) => prev + 5);
+          }
+        }
+      }
+    }, 200),
+    [visibleCount, activities.length]
+  );
+
+  useEffect(() => {
+    const container = containerRef.current;
+    if (container) {
+      container.addEventListener('scroll', handleScroll);
+    }
+
+    // Cleanup event listener on unmount
+    return () => {
+      handleScroll.cancel(); // Cancel any pending debounced calls
+      if (container) {
+        container.removeEventListener('scroll', handleScroll);
+      }
+    };
+  }, [handleScroll]); // Only re-run when handleScroll changes
+  return (
+    <Card className="w-full border-none rounded-[8px] shadow-sm">
+      <div ref={containerRef} className="px-12 py-8 h-[500px] overflow-y-auto scrollbar-hide">
+        {/* Add a style block to hide WebKit scrollbars */}
+        <style>{`
+          div::-webkit-scrollbar {
```

Comment:
Please handle this styling separately using Tailwind CSS if possible. Otherwise, create a new activity-log-timeline.css file and add the styles there.